### PR TITLE
Snow 1515844 init integration tests; incorrect URL error handling

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
* Add `snow init` integration tests
* Add readable error message in case chosen URL does not exist

### CODE REVIEW

File: src/snowflake/cli/plugins/init/commands.py [hunk 1]
```diff
@@ -98,13 +98,22 @@ def _fetch_remote_template(
...
-    shallow_git_clone(url, to_path=destination.path)
+    try:
+        shallow_git_clone(url, to_path=destination.path)
+    except GitCommandError as err:
+        import re
+
+        if re.search("fatal: repository '.*' not found", err.stderr):
+            raise ClickException(f"Repository '{url}' does not exist")
+        raise
+
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/plugins/init/commands.py:116]: Sneaky improvement 😄

File: tests_integration/test_init.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    assert _list_contents(Path(local_path)) == _list_contents(Path(local_path))
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("template", ["example_snowpark", "example_streamlit"])
+def test_default_url(runner, temporary_working_directory, template):
+    path = "a_new_project"
+    runner = runner.invoke(["init", path, "--template", template, "--no-interactive"])
+    assert runner.exit_code == 0, runner.output
+    assert (Path(path) / "snowflake.yml").exists()
```
Comment (sfc-gh-astus) [inline tests_integration/test_init.py:106]: Is it working for many cases? I think if firsts case pass, the rest of them will pass too.

Comment (sfc-gh-pczajka) [reply tests_integration/test_init.py:106]: Removed multiple cases

File: src/snowflake/cli/plugins/init/commands.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/init/commands.py:116]: Mentioned in PR title as well

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**